inference-engine/ie_bridges/java/cpp/jni_common.hpp

```diff
@@ -0,0 +1,101 @@
+    {
+        const char *utf_str = env->GetStringUTFChars(jstr, 0);
+        std::string n_str(utf_str ? utf_str : "");
+        env->ReleaseStringUTFChars(jstr, utf_str);
+        return n_str;
+    }
+    catch (const std::exception &e)
+    {
+        throwJavaException(env, &e, method_name);
+    }
+    catch (...)
+    {
+        throwJavaException(env, 0, method_name);
+    }
+    return nullptr;
+}
+
+static std::map<std::string, std::string> javaMapToMap(JNIEnv *env, jobject java_map)
+{
+    static const char method_name[] = "javaMapToMap";
+    try
+    {
+        jclass map_class = env->GetObjectClass(java_map);
+        jmethodID entrySet_method_id = env->GetMethodID(map_class, "entrySet", "()Ljava/util/Set;");
+        jobject entry_set = env->CallObjectMethod(java_map, entrySet_method_id);
+        
+        jclass set_class = env->FindClass("java/util/Set");
+        jmethodID iterator_method_id = env->GetMethodID(set_class, "iterator", "()Ljava/util/Iterator;");
+        jobject iterator = env->CallObjectMethod(entry_set, iterator_method_id);
+        
+        jclass iterator_class = env->FindClass("java/util/Iterator");
+        jmethodID hasNext_method_id = env->GetMethodID(iterator_class, "hasNext", "()Z");
+        jmethodID next_method_id = env->GetMethodID(iterator_class, "next", "()Ljava/lang/Object;");
+        bool hasNext = (bool) (env->CallBooleanMethod(iterator, hasNext_method_id) == JNI_TRUE);
+
+        jclass mapentry_class = (env)->FindClass("java/util/Map$Entry");
+        jmethodID get_key_method_id = env->GetMethodID(mapentry_class, "getKey", "()Ljava/lang/Object;");
+        jmethodID get_value_method_id = env->GetMethodID(mapentry_class, "getValue", "()Ljava/lang/Object;");
+
+        jclass string_class = env->FindClass("java/lang/String");
+        jmethodID to_string_method_id = env->GetMethodID(string_class, "toString", "()Ljava/lang/String;");
+
+        std::map<std::string, std::string> c_map;
+
+        while(hasNext) {
+		    jobject entry = env->CallObjectMethod(iterator, next_method_id);
+
+		    jstring key = (jstring) env->CallObjectMethod(env->CallObjectMethod(entry, get_key_method_id), to_string_method_id);
+		    jstring value = (jstring) env->CallObjectMethod(env->CallObjectMethod(entry, get_value_method_id), to_string_method_id);    
+
+            c_map.insert(std::make_pair(jstringToString(env, key), jstringToString(env, value)));
+
+		    hasNext = (bool) (env->CallBooleanMethod(iterator, hasNext_method_id) == JNI_TRUE);
+	    }
+
+        return c_map;
+    }
+    catch (const std::exception &e)
+    {
+        throwJavaException(env, &e, method_name);
+    }
+    catch (...)
+    {
+        throwJavaException(env, 0, method_name);
+    }
+    return std::map<std::string, std::string>();
+}
+
+template<typename Key, typename Val>
+static Key find_by_value(std::map<Key, Val>& map, Val& value) {
+    for (auto &itr : map)
+        if (itr.second == value)
+            return itr.first;
```

Comment:
Why loop? Can we use `.find`?

Reply:
.find finds an element by key, but here we should find key by value